File: aeo-platform/backend/app/schemas/session.py
"""Session schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.session import SessionStatus


class SessionBase(BaseModel):
    """Base session schema."""

    title: str | None = None
    status: SessionStatus = SessionStatus.ACTIVE
    entity_id: UUID | None = None
    context: str | None = None
    extra_metadata: str | None = None


class SessionCreate(SessionBase):
    """Schema for creating a session."""

    pass


class SessionUpdate(BaseModel):
    """Schema for updating a session."""

    title: str | None = None
    status: SessionStatus | None = None
    entity_id: UUID | None = None
    context: str | None = None
    extra_metadata: str | None = None


class SessionResponse(SessionBase):
    """Schema for session response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    created_at: datetime
    updated_at: datetime
    completed_at: datetime | None


class SessionListItem(BaseModel):
    """会话列表单项 schema。"""

    id: UUID
    title: str | None = None
    status: SessionStatus
    entity_id: UUID | None = None
    brand_name: str | None = None
    last_message_preview: str | None = None
    message_count: int = 0
    created_at: datetime
    updated_at: datetime


class SessionListResponse(BaseModel):
    """会话列表响应 schema。"""

    sessions: list[SessionListItem]
    total: int

File: frontend/src/types/session.ts
export interface Session {
  id: string;
  createdAt: Date;
  updatedAt: Date;
  brandName?: string;
  status: 'active' | 'archived';
}

export interface SessionCreate {
  brandName?: string;
}

export interface SessionListItem {
  id: string;
  title: string | null;
  status: string;
  brand_name: string | null;
  entity_id: string | null;
  last_message_preview: string | null;
  message_count: number;
  created_at: string;
  updated_at: string;
}

export interface SessionListResponse {
  sessions: SessionListItem[];
  total: number;
}
